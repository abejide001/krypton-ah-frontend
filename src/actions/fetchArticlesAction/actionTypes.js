const actionTypes = {
  FETCH_POPULAR_ARTICLES_SUCCESS: 'FETCH_POPULAR_ARTICLES_SUCCESS',
  FETCH_TRENDING_ARTICLES_SUCCESS: 'FETCH_TRENDING_ARTICLES_SUCCESS',
  FETCH_CATEGORIES_SUCCESS: 'FETCH_CATEGORIES_SUCCESS',
  FETCH_FEATURED_ARTICLES_SUCCESS: 'FETCH_FEATURED_ARTICLES_SUCCESS',
  FETCH_POPULAR_ARTICLES_FAILURE: 'FETCH_POPULAR_ARTICLES_FAILURE',
  FETCH_TRENDING_ARTICLES_FAILURE: 'FETCH_TRENDING_ARTICLES_FAILURE',
  FETCH_CATEGORIES_FAILURE: 'FETCH_CATEGORIES_FAILURE',
  FETCH_TAGS_SUCCESS: 'FETCH_TAGS_SUCCESS',
  FETCH_TAGS_FAILURE: 'FETCH_TAGS_FAILURE',
  FETCH_FEATURED_ARTICLES_FAILURE: 'FETCH_FEATURED_ARTICLES_FAILURE'
};

export default actionTypes;
