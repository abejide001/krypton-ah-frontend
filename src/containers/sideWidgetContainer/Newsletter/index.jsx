import React from 'react';
import { Grid, Input, Button } from 'semantic-ui-react';
import Title from '../../../components/sectionHeader';

export default function NewsLetter() {
  return (
    <Grid.Column className="NewsLetter">
      <Title text="NewsLetter" />
      <Input placeholder="Subscribe..." />
      <Button basic color="black">
        Subsrcibe
      </Button>
    </Grid.Column>
  );
}
