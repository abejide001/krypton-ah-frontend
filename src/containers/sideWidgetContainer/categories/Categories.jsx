import React from 'react';
import { Grid, List, Label } from 'semantic-ui-react';
import Title from '../../../components/sectionHeader';
import Tags from '../../../components/tags';

export default function categories({ category }) {
  return (
    <Grid.Column className="Categories">
      <Title text="Categories" />
      <List divided relaxed>
        <Tags tags={category} />
        <List.Item>
          <Label as="a">more</Label>
        </List.Item>
      </List>
    </Grid.Column>
  );
}
