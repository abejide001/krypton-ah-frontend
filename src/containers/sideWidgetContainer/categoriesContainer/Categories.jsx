import React from 'react';
import PropTypes from 'prop-types';
import { Grid, List, Label } from 'semantic-ui-react';
import Tags from '../../../components/tags';

export default function categories({ category }) {
  return (
    <Grid.Column className="Categories">
      <List divided relaxed>
        <Tags tags={category} />
        <List.Item>
          <Label as="a">more</Label>
        </List.Item>
      </List>
    </Grid.Column>
  );
}

categories.defaultProps = {
  category: []
};

categories.propTypes = {
  category: PropTypes.array
};
