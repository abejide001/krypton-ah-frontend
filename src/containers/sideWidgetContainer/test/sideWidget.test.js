import React from 'react';
import { shallow } from 'enzyme';
import { Widget } from '../index';
import Newsletter from '../Newsletter';
import Categories from '../categoriesContainer/Categories';
import { fetchcategories, fetchtags } from '../../../actions/fetchArticlesAction/fetchArticlesActions';

describe('button test', () => {
  it('should test for the sidewidget component', () => {
    const wrapper = shallow(<Widget fetchcategories={fetchcategories} fetchtags={fetchtags} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the newsletter component', () => {
    const wrapper = shallow(<Newsletter />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the categories component', () => {
    const wrapper = shallow(<Categories />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the categories component', () => {
    const wrapper = shallow(<Categories />);
    expect(wrapper).toMatchSnapshot();
  });
});
