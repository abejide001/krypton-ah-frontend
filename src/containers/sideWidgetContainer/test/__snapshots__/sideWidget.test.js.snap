// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test should test for the categories component 1`] = `
<GridColumn
  className="Categories"
>
  <List
    divided={true}
    relaxed={true}
  >
    <Tags
      tags={Array []}
    />
    <ListItem>
      <Label
        as="a"
      >
        more
      </Label>
    </ListItem>
  </List>
</GridColumn>
`;

exports[`button test should test for the newsletter component 1`] = `
<GridColumn
  className="NewsLetter"
>
  <sectionHeader
    text="NewsLetter"
  />
  <Input
    placeholder="Search..."
    type="text"
  />
  <Button
    as="button"
    basic={true}
    color="black"
  >
    Submit
  </Button>
</GridColumn>
`;

exports[`button test should test for the sidewidget component 1`] = `
<GridRow
  columns={1}
>
  <GridColumn>
    <sectionHeader
      text="Trending"
    />
    <ArticlesFeedContainer
      articles={
        Array [
          Object {
            "content": "Apple's news at CES 2019 shows it's facing some hard truths",
            "description": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ",
            "title": "Trending Title",
          },
          Object {
            "content": "Apple's news at CES 2019 shows it's facing some hard truths",
            "description": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ",
            "title": "Trending Title",
          },
          Object {
            "content": "Apple's news at CES 2019 shows it's facing some hard truths",
            "description": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ",
            "title": "Trending Title",
          },
          Object {
            "content": "Apple's news at CES 2019 shows it's facing some hard truths",
            "description": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ",
            "title": "Popular Title",
          },
          Object {
            "content": "Apple's news at CES 2019 shows it's facing some hard truths",
            "description": "When bobbing for apples, an idealist endlessly reaches for the best apple, a pessimist ",
            "title": "Popular Title",
          },
        ]
      }
      className="featured-widget"
    />
    <sectionHeader
      text="Tags"
    />
    <List
      className="tags"
      divided={true}
      relaxed={true}
    >
      <Tags />
    </List>
    <categories
      category={Array []}
    />
  </GridColumn>
  <index />
  <NewsLetter />
</GridRow>
`;
